src/main/resources/assets/js/application/recruiter/services/tdprRecruiterSlotsService.js

```diff
@@ -0,0 +1,54 @@
+define(['angular', 'application/recruiter/tdprRecruiterModule', 'application/recruiter/services/tdprDateService'], function (angular, tdprRecruiterModule) {
+    tdprRecruiterModule.service('tdprRecruiterSlotsService', ['$http', '$q', 'tdprDateService', 'dateFilter', 'AvailabilityEnum', function ($http, $q, tdprDateService, dateFilter, AvailabilityEnum) {
+        var updateSlots = function (slots, personId, startDate, endDate) {
+            return $http.put("/api/slots/update/" + personId + "/" + startDate + "/" + endDate, slots).then(function (response) {
+                return response;
+            }, function (err) {
+                err.message = "Failed to update slots for person.";
+                return $q.reject(err);
+            });
+        };
+
+        var reformatSlots = function (slots, day, personId) {
+            var array = [];
+            var dateObj = tdprDateService.resetDate(day);
+
+            for (var key in slots) {
+                if (!slots.hasOwnProperty(key)) continue;
+
+                var compareDay = tdprDateService.resetDate(slots[key].day);
+
+                var slotType = slots[key].type ? slots[key].type : "empty";
+
+                if (compareDay.getTime() === dateObj.getTime()) {
+                    array.push({
+                        slotsDate: tdprDateService.resetDate(compareDay),
+                        person: {id: personId},
+                        slot: {id: slots[key].slot},
+                        type: {id: AvailabilityEnum[slotType].id}
+                    });
+                }
+            }
+            return array;
+        };
+
+        this.reformatSlots = function (slots, day, personId) {
```

Comment:
To work with already existing end-point. Slots we are getting from one endpoint differs from which another end-point expects. It might get changed with refactor, but that's what we ended up doing.

This is what endpoint send us:
{
    "day": "2016-08-10",
    "id": 45146,
    "person": 9,
    "slot": 7,
    "type": "available"
}

This is what is expected on back-end for slots to update:
{
    "person": {
        "id": 11
    },
    "slot": {
        "id": 7
    },
    "slotsDate": "2016-08-08T00:00:00.000Z",
    "type": {
        "id": 2
    }
}